### ISSUE TITLE
P2P sending attachments success with 502 incomplete MIME error 

### ISSUE THREAD

**Issue by pasin** [couchbase/couchbase-lite-ios#770]
This happens only when enabling authentication and pushing documents with attachments. The operation was success but there is an error shown up on both listener and replicator log. Below is a log file from the listener side (photo-drop):

```
13:32:32.781| Couchbase Lite using SQLite version 3.8.8 (2015-01-16 12:24:00 518621f8dcd6c0cc107110991753731a19b263f6)
13:32:32.782| FYI, couldn't enable SQLite mmap
13:32:33.305‖ HTTPServer: Started HTTP server on port 64254
13:32:40.631‖ CBLListener: Login attempted for user '%2F+GPrrBo17DPacTmPWAcGRh+baQUnqOC9vaD4bipg5Q='
13:32:40.631‖ CBLListener: GET /db/_local/ad8b6de472f7b767c305c0ed83e33566c9c1197b
13:32:40.633‖ CBLListener:     Response[GET /db/_local/ad8b6de472f7b767c305c0ed83e33566c9c1197b] --> 404
13:32:40.648‖ CBLListener: Login attempted for user '%2F+GPrrBo17DPacTmPWAcGRh+baQUnqOC9vaD4bipg5Q='
13:32:40.648‖ CBLListener: POST /db/_revs_diff {+66}
13:32:40.649‖ CBLListener:     Response[POST /db/_revs_diff] --> 200
13:32:42.212‖ CBLListener: Login attempted for user '%2F+GPrrBo17DPacTmPWAcGRh+baQUnqOC9vaD4bipg5Q='
13:32:42.212‖ CBLListener: PUT /db/-8CyTcNboCpbyyiAYBHgAMQ?new_edits=false {+1466092}
13:32:42.241‖ SyncVerbose: CBLMultipartDocumentReader[_id="(null)"]: has attachments, multipart/related; boundary="-KolyjHoLVdV47FB-AHRz-u"
13:32:42.243‖ SyncVerbose: CBLMultipartDocumentReader[_id="(null)"]: Finished loading (0 attachments)
13:32:42.243‖ WARNING: CBLMultipartDocumentReader[_id="(null)"]: received incomplete MIME multipart response
13:32:42.244‖ CBLListener:     Response[PUT /db/-8CyTcNboCpbyyiAYBHgAMQ] --> 502
13:32:46.916‖ CBLListener: Login attempted for user '%2F+GPrrBo17DPacTmPWAcGRh+baQUnqOC9vaD4bipg5Q='
13:32:46.917‖ CBLListener: PUT /db/-8CyTcNboCpbyyiAYBHgAMQ?new_edits=false {+1466092}
13:32:46.918‖ SyncVerbose: CBLMultipartDocumentReader[_id="(null)"]: has attachments, multipart/related; boundary="-QTe9-g1wYYsc1OeGUp0ENM"
13:32:46.920‖ SyncVerbose: CBLMultipartDocumentReader[_id="-8CyTcNboCpbyyiAYBHgAMQ"]: Starting attachment #1...
13:32:46.921‖ CBL_BlobStore /Users/pasin/Library/Developer/CoreSimulator/Devices/07502C96-0383-4FFB-946B-54CE975845B6/data/Containers/Data/Application/646C8776-3112-40AE-AFE3-09BE51FEBD7E/Library/Application Support/CouchbaseLite/db.cblite2/attachments created tempDir /Users/pasin/Library/Developer/CoreSimulator/Devices/07502C96-0383-4FFB-946B-54CE975845B6/data/Containers/Data/Application/646C8776-3112-40AE-AFE3-09BE51FEBD7E/Library/Application Support/CouchbaseLite/db.cblite2/(A Document Being Saved By PhotoDrop)
13:32:46.934‖ SyncVerbose: CBLMultipartDocumentReader[_id="-8CyTcNboCpbyyiAYBHgAMQ"]: Finished attachment #1: len=1431k, digest=md5-lCTTeEf/08tnTFgXLZoh+w==, SHA1=<03357f5a 20091c22 580cc2d1 cfc9841c e2f485ec>
13:32:46.934‖ SyncVerbose: CBLMultipartDocumentReader[_id="-8CyTcNboCpbyyiAYBHgAMQ"]: Finished loading (1 attachments)
13:32:46.937‖ CBLListener:     Response[PUT /db/-8CyTcNboCpbyyiAYBHgAMQ] --> 201
13:32:46.970‖ CBLListener: Login attempted for user '%2F+GPrrBo17DPacTmPWAcGRh+baQUnqOC9vaD4bipg5Q='
13:32:46.970‖ CBLListener: PUT /db/_local/ad8b6de472f7b767c305c0ed83e33566c9c1197b {+20}
13:32:46.970‖ SyncVerbose: CBLMultipartDocumentReader[_id="(null)"]: Finished loading (0 attachments)
13:32:46.971‖ CBLListener:     Response[PUT /db/_local/ad8b6de472f7b767c305c0ed83e33566c9c1197b] --> 201
```

**Comment by snej** [comment]
The warning is: `WARNING: CBLMultipartDocumentReader[_id="(null)"]: received incomplete MIME multipart response`, and the request URI is `/db/-8CyTcNboCpbyyiAYBHgAMQ?new_edits=false`.

It's not clear whether the error is on the sending or the receiving side.

**Comment by pasin** [comment]
I couldn't reproduce the issue anymore and the replication with attachments and auth enabled seemed much faster than before. I guessed the issue was fixed in a8fb03dbea7533b11cfa18df7456153c67a4644b, which also fixed the new stream when the -connection:needNewBodyStream delegate method is called.

**Comment by snej** [comment]
Yes, I fixed it but forgot that there had been an issue filed on it already. Thanks!
